sky/packages/sky/lib/src/rendering/shifted_box.dart

```diff
@@ -194,8 +194,8 @@ class RenderPositionedBox extends RenderShiftedBox {
     markNeedsLayout();
   }
 
-  bool get _shinkWrapWidth => _shrinkWrap == ShrinkWrap.width || _shrinkWrap == ShrinkWrap.both;
-  bool get _shinkWrapHeight => _shrinkWrap == ShrinkWrap.height || _shrinkWrap == ShrinkWrap.both;
+  bool get _shinkWrapWidth => _shrinkWrap == ShrinkWrap.width || _shrinkWrap == ShrinkWrap.both || (constraints.maxWidth == double.INFINITY);
```

Comment:
add a comment saying that these are only valid during performLayout() and paint(), since they rely on `constraints` which is only set after layout() is called.